roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+"""
+Check that the SDN is routing traffic properly.
+"""
+
+import base64
+import os
+import re
+
+import yaml
+from ansible.module_utils import six
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class SDNCheck(NotContainerizedMixin, OpenShiftCheck):
+    """A check to run relevant diagnostics on the SDN."""
+
+    name = 'sdn'
+    tags = ['health']
+
+    def is_active(self):
+        """Skip hosts that are not masters or nodes."""
+        group_names = self.get_var('group_names', default=[])
+        master_or_node = 'oo_masters_to_config' in group_names or \
+                         'oo_nodes_to_config' in group_names
+        return super(SDNCheck, self).is_active() and master_or_node
+
+    def run(self):
+        if self.want_full_results:
+            # Gather diagnostic information and perform diagnostics on a master
+            # or node host.
+            try:
+                # Boot log can take too long to load on some systems.
+                # self.save_command_output('journal-full.log', ['journalctl',
```

Comment:
So one possible answer to my question about this vs `oc adm diagnostics` would be that we expect users to use `oc adm diagnostics` when their cluster is working at least well enough that that is able to work (eg, nodes can create pods, etc), and we only expect them to use the ansible health check when things are so bad that `oc adm diagnostics` won't work. In that case, getting the full logs is more important, and also potentially less burdensome, since we'd presumably only be running on newly-installed machines...